src/librustc/hir/lowering.rs

```diff
@@ -79,6 +80,8 @@ pub struct LoweringContext<'a> {
     impl_items: BTreeMap<hir::ImplItemId, hir::ImplItem>,
     bodies: FxHashMap<hir::BodyId, hir::Body>,
 
+    loop_scopes: Vec<NodeId>,
```

Comment:
I don't think it needs to be `None`, since in both of the cases its used, its absence will be obvious:
- In the unlabeled case, `.last()` will return `None`
- In the labeled case, the label will not be found
Both of these result in `DUMMY_NODE_ID`.